mission/simulator/platform-services-impl/src/main/java/esa/mo/platform/impl/util/PlatformServicesProviderSoftSim.java

```diff
@@ -75,82 +79,85 @@ public void init(COMServicesProvider comServices) throws MALException {
     GPSAdapterInterface gpsAdapter;
     OpticalDataReceiverAdapterInterface optRxAdapter;
     SoftwareDefinedRadioAdapterInterface sdrAdapter;
-    PowerControlAdapterInterface pcAdapter;
 
     Properties platformProperties = new Properties();
     try {
       platformProperties
           .load(new FileInputStream("platformsim.properties"));
+      String pcAdapterName = platformProperties.getProperty("pc.adapter");
       if (platformProperties.getProperty("platform.mode").equals("hybrid")) {
         String camAdapterName = platformProperties.getProperty("camera.adapter");
         String adcsAdapterName = platformProperties.getProperty("adcs.adapter");
         String gpsAdapterName = platformProperties.getProperty("gps.adapter");
         String optRxAdapterName = platformProperties.getProperty("optrx.adapter");
         String sdrAdapterName = platformProperties.getProperty("sdr.adapter");
-        String pcAdapterName = platformProperties.getProperty("pc.adapter");
+
         try {
-          camAdapter = (CameraAdapterInterface) Class.forName(camAdapterName).newInstance();
+          pcAdapter = (PowerControlAdapterInterface) Class.forName(pcAdapterName).newInstance();
         } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+          Logger.getLogger(PlatformServicesProviderSoftSim.class.getName()).log(Level.WARNING,
+                  "Failed to instantiate the power control adapter.", e);
+          pcAdapter = new PowerControlSoftSimAdapter(instrumentsSimulator);
+        }
+        try {
+          //TODO the old code did not use the esasimulator either, just the default constructor...?!?
+          Constructor constructor = Class.forName(camAdapterName).getConstructor(ESASimulator.class, PowerControlAdapterInterface.class);
```

Comment:
Note that this will fail for hybrid simulation with OPS-SAT adapter - its constructor takes other arguments. Is it possible to fix this and other adapter instantiations?

Reply:
Done. Had to change the OPSSAT NanomindServicesConsumer for this, as OPSSAT adapters were using it but it is very OPSSAT specific.